cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -58,45 +62,106 @@ def proctored_exam_settings(request, course_id):
             "course_start_date": "2013-02-05T05:00:00Z",
             "is_staff": true
         }
+
+    ------------------------------------------------------------------------------------
+    POST
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a set of course proctored exam settings.
+        * 400: Bad Request - Unable to save requested settings.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the course.
+        * 404: The requested course does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will echo the updated proctored
+        exam settings data.
     """
-    course_key = CourseKey.from_string(course_id)
-    with modulestore().bulk_operations(course_key):
-        if request.method == 'GET':
-            course_module = get_course_and_check_access(course_key, request.user)
+    PROCTORED_EXAM_SETTINGS_KEYS = [
+        'enable_proctored_exams',
+        'allow_proctoring_opt_out',
+        'proctoring_provider',
+        'proctoring_escalation_email',
+        'create_zendesk_tickets',
+    ]
+
+    def get(self, request, course_id):
+        """ GET handler """
+        course_module = self._get_and_validate_course(request.user, course_id)
```

Comment:
Is there a reason you removed the `modulestore().bulk_operations` context manager here?

Reply:
seems unneeded since this function is only reading right?

Reply:
It seems to be used regardless of whether reads or write are occurring throughout the code base. The original advanced_settings_handler does it [here](https://github.com/edx/edx-platform/blob/7ea9714a3385123253304da340c23bfad97e2cd2/cms/djangoapps/contentstore/views/course.py#L1321). My understanding is that it should wrap any Mongo intensive operations, even if they're just reads (see this [PR](https://github.com/edx/edx-platform/pull/5370)). I'd err on the side of using it unless we are confident we don't need it.

Here are a few more instances:
https://github.com/edx/edx-platform/blob/7ea9714a3385123253304da340c23bfad97e2cd2/cms/djangoapps/contentstore/views/course.py#L261
https://github.com/edx/edx-platform/blob/7ea9714a3385123253304da340c23bfad97e2cd2/cms/djangoapps/contentstore/views/course.py#L261

Reply:
ok yeah I've just added it in to be safe